// eauth/esm2015/lib/eauth.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/eauth.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import * as i0 from "@angular/core";
export class EauthService {
    constructor() { }
    /**
     * @param {?} str
     * @return {?}
     */
    encrypt(str) {
        /** @type {?} */
        let key = CryptoJS.enc.Utf8.parse("elprosystem");
        /** @type {?} */
        let iv = CryptoJS.enc.Utf8.parse("elprosystem");
        /** @type {?} */
        let encrypted = CryptoJS.AES.encrypt(str, key, {
            iv: iv,
            mode: CryptoJS.mode.CBC,
            padding: CryptoJS.pad.Pkcs7
        });
        return encrypted.toString();
    }
}
EauthService.decorators = [
    { type: Injectable, args: [{
                providedIn: "root"
            },] }
];
/** @nocollapse */
EauthService.ctorParameters = () => [];
/** @nocollapse */ EauthService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function EauthService_Factory() { return new EauthService(); }, token: EauthService, providedIn: "root" });
