// eauth/fesm5/eauth.js
import { Injectable, ɵɵdefineInjectable, Component, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * Generated from: lib/eauth.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var EauthService = /** @class */ (function () {
    function EauthService() {
    }
    /**
     * @param {?} str
     * @return {?}
     */
    EauthService.prototype.encrypt = /**
     * @param {?} str
     * @return {?}
     */
    function (str) {
        /** @type {?} */
        var key = CryptoJS.enc.Utf8.parse("elprosystem");
        /** @type {?} */
        var iv = CryptoJS.enc.Utf8.parse("elprosystem");
        /** @type {?} */
        var encrypted = CryptoJS.AES.encrypt(str, key, {
            iv: iv,
            mode: CryptoJS.mode.CBC,
            padding: CryptoJS.pad.Pkcs7
        });
        return encrypted.toString();
    };
    EauthService.decorators = [
        { type: Injectable, args: [{
                    providedIn: "root"
                },] }
    ];
    /** @nocollapse */
    EauthService.ctorParameters = function () { return []; };
    /** @nocollapse */ EauthService.ngInjectableDef = ɵɵdefineInjectable({ factory: function EauthService_Factory() { return new EauthService(); }, token: EauthService, providedIn: "root" });
    return EauthService;
}());

/**
 * @fileoverview added by tsickle
 * Generated from: lib/eauth.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var EauthComponent = /** @class */ (function () {
    function EauthComponent() {
    }
    /**
     * @return {?}
     */
    EauthComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    EauthComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-eauth',
                    template: "\n    <p>\n      eauth works!\n    </p>\n  "
                }] }
    ];
    /** @nocollapse */
    EauthComponent.ctorParameters = function () { return []; };
    return EauthComponent;
}());

/**
 * @fileoverview added by tsickle
 * Generated from: lib/eauth.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var EauthModule = /** @class */ (function () {
    function EauthModule() {
    }
    EauthModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [EauthComponent],
                    imports: [],
                    exports: [EauthComponent]
                },] }
    ];
    return EauthModule;
}());

/**
 * @fileoverview added by tsickle
 * Generated from: index.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * Generated from: eauth.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { EauthComponent, EauthModule, EauthService };
