// eauth/esm5/lib/eauth.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/eauth.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import * as i0 from "@angular/core";
var EauthService = /** @class */ (function () {
    function EauthService() {
    }
    /**
     * @param {?} str
     * @return {?}
     */
    EauthService.prototype.encrypt = /**
     * @param {?} str
     * @return {?}
     */
    function (str) {
        /** @type {?} */
        var key = CryptoJS.enc.Utf8.parse("elprosystem");
        /** @type {?} */
        var iv = CryptoJS.enc.Utf8.parse("elprosystem");
        /** @type {?} */
        var encrypted = CryptoJS.AES.encrypt(str, key, {
            iv: iv,
            mode: CryptoJS.mode.CBC,
            padding: CryptoJS.pad.Pkcs7
        });
        return encrypted.toString();
    };
    EauthService.decorators = [
        { type: Injectable, args: [{
                    providedIn: "root"
                },] }
    ];
    /** @nocollapse */
    EauthService.ctorParameters = function () { return []; };
    /** @nocollapse */ EauthService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function EauthService_Factory() { return new EauthService(); }, token: EauthService, providedIn: "root" });
    return EauthService;
}());
export { EauthService };
