// eauth/fesm2015/eauth.js
import { Injectable, ɵɵdefineInjectable, Component, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * Generated from: lib/eauth.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class EauthService {
    constructor() { }
    /**
     * @param {?} str
     * @return {?}
     */
    encrypt(str) {
        /** @type {?} */
        let key = CryptoJS.enc.Utf8.parse("elprosystem");
        /** @type {?} */
        let iv = CryptoJS.enc.Utf8.parse("elprosystem");
        /** @type {?} */
        let encrypted = CryptoJS.AES.encrypt(str, key, {
            iv: iv,
            mode: CryptoJS.mode.CBC,
            padding: CryptoJS.pad.Pkcs7
        });
        return encrypted.toString();
    }
}
EauthService.decorators = [
    { type: Injectable, args: [{
                providedIn: "root"
            },] }
];
/** @nocollapse */
EauthService.ctorParameters = () => [];
/** @nocollapse */ EauthService.ngInjectableDef = ɵɵdefineInjectable({ factory: function EauthService_Factory() { return new EauthService(); }, token: EauthService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * Generated from: lib/eauth.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class EauthComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
EauthComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-eauth',
                template: `
    <p>
      eauth works!
    </p>
  `
            }] }
];
/** @nocollapse */
EauthComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * Generated from: lib/eauth.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class EauthModule {
}
EauthModule.decorators = [
    { type: NgModule, args: [{
                declarations: [EauthComponent],
                imports: [],
                exports: [EauthComponent]
            },] }
];

/**
 * @fileoverview added by tsickle
 * Generated from: index.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * Generated from: eauth.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { EauthComponent, EauthModule, EauthService };
