// eauth/bundles/eauth.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('eauth', ['exports', '@angular/core'], factory) :
    (global = global || self, factory(global.eauth = {}, global.ng.core));
}(this, (function (exports, core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/eauth.service.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var EauthService = /** @class */ (function () {
        function EauthService() {
        }
        /**
         * @param {?} str
         * @return {?}
         */
        EauthService.prototype.encrypt = /**
         * @param {?} str
         * @return {?}
         */
        function (str) {
            /** @type {?} */
            var key = CryptoJS.enc.Utf8.parse("elprosystem");
            /** @type {?} */
            var iv = CryptoJS.enc.Utf8.parse("elprosystem");
            /** @type {?} */
            var encrypted = CryptoJS.AES.encrypt(str, key, {
                iv: iv,
                mode: CryptoJS.mode.CBC,
                padding: CryptoJS.pad.Pkcs7
            });
            return encrypted.toString();
        };
        EauthService.decorators = [
            { type: core.Injectable, args: [{
                        providedIn: "root"
                    },] }
        ];
        /** @nocollapse */
        EauthService.ctorParameters = function () { return []; };
        /** @nocollapse */ EauthService.ngInjectableDef = core.ɵɵdefineInjectable({ factory: function EauthService_Factory() { return new EauthService(); }, token: EauthService, providedIn: "root" });
        return EauthService;
    }());

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/eauth.component.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var EauthComponent = /** @class */ (function () {
        function EauthComponent() {
        }
        /**
         * @return {?}
         */
        EauthComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () {
        };
        EauthComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'lib-eauth',
                        template: "\n    <p>\n      eauth works!\n    </p>\n  "
                    }] }
        ];
        /** @nocollapse */
        EauthComponent.ctorParameters = function () { return []; };
        return EauthComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/eauth.module.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var EauthModule = /** @class */ (function () {
        function EauthModule() {
        }
        EauthModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [EauthComponent],
                        imports: [],
                        exports: [EauthComponent]
                    },] }
        ];
        return EauthModule;
    }());

    exports.EauthComponent = EauthComponent;
    exports.EauthModule = EauthModule;
    exports.EauthService = EauthService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
